devito/passes/iet/dtypes.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+
+from devito.arch.compiler import Compiler
+from devito.ir import Callable, FindSymbols, SymbolRegistry
+from devito.passes.iet.engine import iet_pass
+from devito.passes.iet.langbase import LangBB
+from devito.tools import as_tuple
+
+__all__ = ['lower_dtypes']
+
+
+@iet_pass
+def _complex_includes(iet: Callable, lang: type[LangBB], compiler: Compiler,
+                      sregistry: SymbolRegistry) -> tuple[Callable, dict]:
+    """
+    Includes complex arithmetic headers for the given language, if needed.
+    """
+
+    # Check if there are complex numbers that always take dtype precedence
+    is_complex = False
+    for f in FindSymbols().visit(iet):
```

Comment:
a bit easier to read and more pythonic:

```
for f in FindSymbols().visit(iet):
    try:
        if np.issubdtype(f.dtype, np.complexfloating):
            break
    except:
        pass
else:
      return iet, {}
```

this way you don't need the `is_complex` flag